frontend/src/metabase/visualizations/components/settings/ChartSettingInputNumeric.tsx

```diff
@@ -14,20 +16,25 @@ const ChartSettingInputNumeric = ({
   value,
   ...props
 }: ChartSettingInputProps) => {
-  const [internalValue, setInternalValue] = useState(value);
+  const [internalValue, setInternalValue] = useState(value?.toString() ?? "");
```

Comment:
The root cause of this bug was that Safari doesn't now allow decimals in inputs with `type="number"`. Below we changed that to `type="text"`, and here we are now storing the internal value as a string to match the change in input type.